litestar/cli/_utils.py

```diff
@@ -10,15 +10,26 @@
 from pathlib import Path
 from typing import TYPE_CHECKING, Any, Callable, Generator, Iterable, Sequence, TypeVar, cast
 
-from click import ClickException, Command, Context, Group, pass_context, style
-from rich.console import Console
+from rich import get_console
 from rich.table import Table
 from typing_extensions import Concatenate, ParamSpec
 
 from litestar import Litestar, __version__
 from litestar.middleware import DefineMiddleware
 from litestar.utils import get_name
 
+if TYPE_CHECKING:
+    from click import ClickException, Command, Context, Group, pass_context
+else:
+    try:  # pragma: no cover
+        from rich_click import ClickException, Context, pass_context
+        from rich_click.rich_command import RichCommand as Command
+        from rich_click.rich_group import RichGroup as Group
+
+    except ImportError:  # pragma: no cover
+        from click import ClickException, Command, Context, Group, pass_context
```

Comment:
```suggestion
rich_click_installed = False
try:
    import rich_click
    rich_click_installed = True
except ImportError:
    pass

if TYPE_CHECKING or not rich_click_installed:
    from click import ClickException, Command, Context, Group, pass_context
else:
    from rich_click import ClickException, Context, pass_context
    from rich_click.rich_command import RichCommand as Command
    from rich_click.rich_group import RichGroup as Group
```

would be a good way to handle this I think. You could even make `rich_click_installed` a constant in `cli/_utils.py` and then reuse it.